src/core/config/reader.rs

```diff
@@ -131,16 +129,16 @@ impl ConfigReader {
                     let meta = self.proto_reader.fetch(link.src.as_str()).await?;
 
                     for m in meta {
-                        config_module.extensions.add_proto(m);
+                        extensions.add_proto(m);
                     }
                 }
             }
         }
 
         // Recreating the ConfigModule in order to recompute the values of
         // `input_types`, `output_types` and `interface_types`
-        let mut final_config_module = ConfigModule::from(config_module.config);
-        final_config_module.extensions = config_module.extensions;
+        let mut final_config_module = ConfigModule::from(config_module.config().clone());
+        final_config_module.with_extensions(extensions);
         Ok(final_config_module)
```

Comment:
```suggestion
        Ok(ConfigModule::from(config).with_extensions(extensions))
```

This is a much better usage. Notice that the `config` passed to the `from` method. It is a config that is extracted in begining of the function and then modified whenever we hit a `LinkType::Config`.

Feel free to connect on discord if it's not clear.